### PR TITLE
#164857200 Initial tests for the authentication app

### PR DESCRIPTION
**Pull request by Tevinthuku**
### What does this PR do?

This PR sets up the initial tests for the authentication app

### Tasks to be completed

- [x] - Initialize tests directory
- [x] - Create registration tests
- [x] - Create login tests

### How can this be manually tested?

- Clone the repo and checkout `ch/164857200-write-auth-tests` branch
- Create and activate a virtual environment
```
$ virtualenv env
$ source env/bin/activate
```
- Install requirements
```
$ pip install -r requirements.txt
```
- Make migrations
```
$ python manage.py makemigrations
```
- Run migrations
```
$ python manage.py migrate
```
- Run the tests
```
python manage.py test
```

### PT stories relevant to this PR?

[#164857200](https://www.pivotaltracker.com/n/projects/2321565/stories/164857200)

### CODE REVIEW

File: authors/apps/authentication/tests/test_login.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            format="json")
+        self.assertEqual(login_user.status_code,
+                         status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            json.loads(login_user.content),
+            {"errors":
+             {"password":
+              ['This field is required.']}})
+
+    def test_user_login_with_invalid_mail(self):
```
Comment (Tevinthuku) [inline authors/apps/authentication/tests/test_login.py:117]: I think this behaviour can be fixed in another PR, I think it should first check if the email is valid then proceed to check if the user has an account or not

Comment (Tevinthuku) [reply authors/apps/authentication/tests/test_login.py:117]: image
I believe this is the root of the problem 
the `email` should be an `EmailField` not a `CharField`

**Review by misocho — COMMENTED** [review]
image
I'm getting the above error. Should it be working this way?

**Comment by Tevinthuku** [comment]
Nope :thinking: They are all passing in my machine. How did you set up the repo so as to run the tests?

image

**Comment by Tevinthuku** [comment]
I think you may be running a python version less than python 3.6. 
Python Versions greater than 3.6 do the decoding from bytes automatically

**Review by justMuriithi — APPROVED** [review]
Screenshot from 2019-03-25 21-49-37
All 14 tests passing

**Review by justMuriithi — APPROVED** [review]
Looks good

**Review by misocho — APPROVED** [review]
Yeah, I was running python3.5, works well on python3.6. Good to go.

**Review by martinMutuma — COMMENTED** [review]
Could we have the base tests for login purposes of other tests to get the token and Other things that will be done for all Tests?

**Comment by Tevinthuku** [comment]
Sure, we can. however, this would mean that no other test files will have a setup method and if they do then the setup method of the BaseTest will be overwritten. 

I can include the Base test file in my other PR since I'm specifically working on `JWT tokens` after login and signup

**Pull request merged**